mpas_analysis/shared/climatology/climatology.py

```diff
@@ -480,11 +494,178 @@ def compute_climatology(ds, monthValues, calendar=None):
 
     climatologyMonths = ds.where(mask, drop=True)
 
-    climatology = _compute_masked_mean(climatologyMonths)
+    climatology = _compute_masked_mean(climatologyMonths, maskVaries)
 
     return climatology
 
 
+def cache_climatologies(ds, monthValues, config, cachePrefix, calendar,
+                        printProgress=False):  # {{{
+    '''
+    Cache NetCDF files for each year of an annual climatology, and then use
+    the cached files to compute a climatology for the full range of years.
+    The start and end years of the climatology are taken from ``config``, and
+    are updated in ``config`` if the data set ``ds`` doesn't contain this
+    full range.
+
+    Note: only works with climatologies where the mask (locations of ``NaN``
+    values) doesn't vary with time.
```

Comment:
Should we comment on cases where this could be a problem, e.g., moving land ice?